pixelator.html

```diff
@@ -1,126 +1,172 @@
+      return alert(`Pixelation factor exceeds maximum of ${maxPixelationFactor}!`);
+    } else if (pixelationFactor < minPixelationFactor) {
+      multiplierInput.value = minPixelationFactor;
+      return alert(`Pixelation factor falls below minimum of ${minPixelationFactor}!`);
+    }
+    draw();
+  })
+
+  sourceImage.addEventListener('load', () => {
+    draw(); // triggered when sourceImage was loaded
+  });
+
+  function draw() {
+    let imageHeight = sourceImage.height;
+    let imageWidth = sourceImage.width;
+    canvas.width = imageWidth;
+    canvas.height = imageHeight;
+
+    // draw centered source image
+    const posX = Math.ceil(canvas.width/2 - imageWidth/2);
+    const posY = Math.ceil(canvas.height/2 - imageHeight/2);
+    ctx.drawImage(sourceImage, posX, posY, imageWidth, imageHeight);
+
+    // read rendered image pixels
+    const imageData = ctx.getImageData(posX, posY, imageWidth, imageHeight);
+    const pixelvalues = imageData.data;
+    // console.log(pixelvalues)
+
+    // draw pixelized image
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    // lööp over image, each 5th pixel
+    let loopkiller = 100000; // around 100k
+
+    for (let row = 1; row < imageHeight; row = row + pixelationFactor) {
+      for (let col = 1; col < imageWidth; col = col + pixelationFactor) {
+        loopkiller--;
+        if (loopkiller < 0) { return console.log("killed"); }
+
+        // get 1st pixel color
+        const pixelReference = ((row*imageWidth)+(col))*4;
+        if (pixelReference === null || typeof pixelReference === "undefined") {return console.log("Invalid pixel reference"); }
+
+        const color = `rgb(
               ${pixelvalues[pixelReference]},
               ${pixelvalues[pixelReference+1]},
               ${pixelvalues[pixelReference+2]}
             )`;
-            // +3 would be the alpha channel, which is pointless for us
-            // console.log(pixelReference, color);
-            // draw "pixel" ball
-            ctx.fillStyle = color;
-            ctx.fillRect(posX + col, posY + row, pixelMultiplier, pixelMultiplier);
-          }
-        }
-
-      });
-
-      btn.addEventListener('click', (e) => {
-        pixelMultiplier = parseInt(in_nbr.value);
-        if (pixelMultiplier > 666) {
-          return alert("thats too much");
-        } else if (pixelMultiplier < 2) {
-          return alert("2 or more pls");
-        }
-
-        const file = in_file.files[0];
-        console.log({ file });
-        if (file.size > 900000) {
-          return alert("Use a smaller image pls");
-        }
-        const fr = new FileReader();
-        fr.onload = () => {
-          img.src = fr.result;
-        };   // onload fires after reading is complete
-        fr.readAsDataURL(file);    // begin reading
-      });
-    </script>
-  </body>
-
+        // +3 would be the alpha channel, which is pointless for us
+        // console.log(pixelReference, color);
+        // draw "pixel" ball
```

Comment:
Oh yeah and these used to be balls, I wanted to do like a particle effect thing and I might still do that later at some point :thinking:

Reply:
I'd love to see that, too!